packages/react-native/React/CxxBridge/RCTCxxBridge.mm

```diff
@@ -1189,86 +1189,88 @@ - (void)dispatchBlock:(dispatch_block_t)block queue:(dispatch_queue_t)queue
 
 - (void)invalidate
 {
-  if (_didInvalidate) {
-    return;
-  }
+  RCTUnsafeExecuteOnMainQueueSync(^{
```

Comment:
This could potentially deadlock. We should not run the unsafe variant of this method. Can you change it with `RCTExecuteOnMainQueue`?

Reply:
Changed.